modules/animation/include/modules/animation/datastructures/basetrack.h

```diff
@@ -264,36 +305,37 @@ auto BaseTrack<Seq>::end() const -> const_iterator {
  *           |-case 2a---|---------case 2b-|
  */
 template <typename Seq>
-void BaseTrack<Seq>::add(Seconds time, bool asNewSequence) {
-    auto addNew = [this](std::unique_ptr<key_type> key) {
+typename BaseTrack<Seq>::key_type* BaseTrack<Seq>::add(Seconds time, bool asNewSequence) {
```

Comment:
use can you trailing return types to avoid having to use the typename thing...
`auto BaseTrack<Seq>::add(Seconds time, bool asNewSequence) -> key_type*`

Reply:
Hmm, true. I both like and dislike that solution. It becomes inconsistent with current code, but on the other side might make it easier for non-template savvy people to understand.

Reply:
Should not the definition be changed as well in that case?

Reply:
since the definition is inside the class decl you can just write 
```
key_type* add(Seconds time, bool asNewSequence);
```
The using decl is available in that scope. It is only when you do the out of class definition that you run into the issue...

Trailing return types are a new thing and we can't really change to use them everywhere, that would be way too much work. But on the other hand they are way to useful in some cases not to use them, hence I think we have to live with having mixed usage. 

No changes needed here for now, just wanted to point out that using trailing return types simplifies this kind of code...

Reply:
Yes, but then the def looks like 
ret func(...) 
and the implementation
auto func(...) -> ret